src/py42/modules/securitydata.py

```diff
@@ -180,56 +184,57 @@ def _search_by_hash(self, hash, type):
     def _find_file_versions(self, md5_hash, sha256_hash):
         file_event_client = self._microservices_client_factory.get_file_event_client()
         pds_client = self._microservices_client_factory.get_preservation_data_service_client()
-        for device_id, paths in file_event_client.get_file_location_detail_by_sha256(sha256_hash):
+        response = file_event_client.get_file_location_detail_by_sha256(sha256_hash)
+        for device_id, paths in parse_file_location_response(response):
             yield pds_client.find_file_versions(md5_hash, sha256_hash, device_id, paths)
 
-    def _stream_file(self, file_generator, file):
+    def _stream_file(self, file_generator):
         for response in file_generator:
             if response.status_code != 200:
                 continue
-            session = self._microservices_client_factory.get_storage_node_session(
-                response[u"storageNodeURL"]
-            )
-            storage_node_client = StorageNodeClient(
-                session,
-                response[u"archiveGuid"],
-                response[u"fileId"],
-                response[u"versionTimestamp"],
-            )
-            token = storage_node_client.get_download_token()
-            return storage_node_client.get_file(response[u"storageNodeURL"], token, file)
-        raise FileNotFoundError("No file available for download.")
+            try:
+                storage_node_client = self._microservices_client_factory.create_storage_preservation_client(
+                    response[u"storageNodeURL"]
+                )
+                token = storage_node_client.get_download_token(
+                    response[u"archiveGuid"], response[u"fileId"], response[u"versionTimestamp"],
+                )
+                return storage_node_client.get_file(str(token))
+            except Py42HTTPError as err:
+                raise_py42_error(err)
```

Comment:
we should `continue` if an error is raised here, that way we only will raise an exception if none of the items in the generator work. If we raise here, this function will exit for the first item in the list that fails.